### PR TITLE
Validate worker node group count

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*
#2657 

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/cluster_defaults.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ func setWorkerNodeGroupDefaults(cluster *Cluster) error {
 		logger.V(1).Info("First worker node group name not specified. Defaulting name to md-0.")
 		cluster.Spec.WorkerNodeGroupConfigurations[0].Name = "md-0"
 	}
+	for i := range cluster.Spec.WorkerNodeGroupConfigurations {
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/cluster_defaults.go:53]:
```suggestion
	for _, wng := range cluster.Spec.WorkerNodeGroupConfigurations {
	    if wng.Count <= 0 {
	      wng.Count = 1
	    }
	
	}
	
```
```suggestion
	for i := range cluster.Spec.WorkerNodeGroupConfigurations {
```

Comment (taneyland) [reply pkg/api/v1alpha1/cluster_defaults.go:53]: I tried the first suggestion initially, but it doesn't actually update the count in the cluster spec. That's why I used the index

**Review by TerryHowe — APPROVED** [review]
/lgtm

Doesn't actually set the worker node group count to 1, but I think this fits the intention

**Comment by taneyland** [comment]
> /lgtm
> 
> Doesn't actually set the worker node group count to 1, but I think this fits the intention

Yes I talked to Joey offline & we decided to just add this validation instead of setting the default to 1. Will update the PR title for clarity

**Review by jiayiwang7 — APPROVED** [review]

**Pull request merged**